actionpack/lib/action_dispatch/system_testing/driver.rb

```diff
@@ -52,8 +56,13 @@ def register_webkit(app)
         end
 
         def setup
+          Capybara.session_name = session_name
           Capybara.current_driver = @name
         end
+
+        def session_name
+          @options.respond_to?(:[]) ? @options[:session_name] || default_session_name : default_session_name
```

Comment:
`.fetch` is not always implemented for instance `Selenium::WebDriver::Remote::W3CCapabilities` support only `[]` method so in order to be more backwards compatible using `[]` and not `fetch`.

Reply:
But is not `options` always a Hash?

Reply:
@rafaelfranca hi sorry for late response... Based on my research it's not true:

From **Capybara**:

https://github.com/teamcapybara/capybara/blob/master/lib/capybara.rb#L477-L482

As you can see they pass as option not hash but instance of:

```ruby
::Selenium::WebDriver::Chrome::Options.new
```

Based on official documentation you can pass more then this (as we do use remote selenium servers e.g. remote capabilities):

https://github.com/SeleniumHQ/selenium/wiki/Ruby-Bindings

```ruby
prefs = {
  prompt_for_download: false, 
  default_directory: "/path/to/dir"
}
options = Selenium::WebDriver::Chrome::Options.new
options.add_preference(:download, prefs)
driver = Selenium::WebDriver.for :chrome, options: options
```
You can also pass an instance of Selenium::WebDriver::Remote::Capabilities, e.g.:

```ruby
caps = Selenium::WebDriver::Remote::Capabilities.internet_explorer(native_events: false)
driver = Selenium::WebDriver.for :remote, desired_capabilities: caps
```